### DISCUSSION TITLE
Does Ring integration show garage door?

### DISCUSSION THREAD

**Discussion by aamays** [dgreif/ring#1349] [Q&A] [answered]
As some of you are aware that myQ has disabled unauthorized third party apps to access their API and the myQ plugin for home bridge has stopped working. I was wondering if the ring plugin for home bridge supports garage door integration that I have added to Ring (since Ring is an authorized third party API). I am able to operate the garage door via the Ring app but I don't see the garage door pop-up as an accessory in my homebridge dashboard.

I could not find documentation anywhere but does the Ring plugin support garage door?

**Comment by tsightler** [comment, marked as answer]
Already discussed in https://github.com/dgreif/ring/issues/1338.  As far as I've been able to tell this is all implemented in the app side, not in the Ring API side, so I don't believe it is possible.

I am a little bit curious though, you say you can "operate the garage door via the Ring app", but everything I've found indicates that you can only get open/close status in the Ring app and any attempt to open/close the door is actually redirecting to the MyQ app.  Is this not true?  Can you provide video?

**Reply by aamays** [reply]
Actually you are right. I lied. I don't know why I remembered operating my garage door via the Ring. But yeah it only shows open and closed status and I can't control the door from Ring app and it just redirects the myQ app. 😅

Such a poor integration. myQ is not even giving proper access to authorized third party apps.

Thanks for clarifying.

**Comment by tsightler** [comment]
> Such a poor integration. myQ is not even giving proper access to authorized third party apps.

To be fair, this isn't uncommon in the industry, and it probably doesn't have anything to do with not giving proper access, it's more just what was the least amount of effort to develop and support while having something to market.  Allowing the Ring app to register for some push notifications and redirecting everything else to the MyQ app is easy and requires very little development or ongoing support for Ring or MyQ's part.  Doing backend integration would require development on both sides as well as ongoing technical coordination for updates, and would potentially significantly complicating supporting a user.  For example, if a user had a problem opening their garage door with the Ring app, they might call Ring, then Ring support would need to engage MyQ support, perhaps involve engineering teams from both sides, etc.  There's just not much in it for either side vs just doing something super simple and marketing it on the website.

I'm going to guess that the MyQ plugin developers will eventually figure out how to behave similarly enough to the MyQ app to not get kicked out.  They will probably have to stop all polling and use push notifications themselves, but that should be possible.  My guess is the polling was driving up the costs of running the service and MyQ decided they needed to stop this.

It wouldn't completely surprise me if Ring eventually implements some of the same anti-reverse engineering techniques as well.  I've already seen some hint of it with some changes in how they are doing authentication in newer versions of the app, but I've always said they could make it really difficult on 3rd-party devs if they wanted to.  When you buy into closed ecosystems you can't be too surprised when the vendor attempts to plug unauthorized holes into that ecosystem.

**Reply by aamays** [reply]
All fair points. I agree that current integration requires minimal/no support on either myq and Ring. For now, I am going with Meross to continue operating garage door via Homekit.
